examples/hallowelt/go_app/README.md

```diff
@@ -0,0 +1,253 @@
+```go
+// Invoke FPC Chaincode retrieveAsset
+	logger.Infof("--> Invoke FPC chaincode retrieveAsset: ")
+	result, err = contract.SubmitTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+```
+* Both previous transactions have been registered on the ledger, to be able to query without this occurring we will use the `EvaluateTransaction` go method instead.
+```go
+	// Evaluate FPC Chaincode retrieveAsset
+	logger.Infof("--> Evaluate FPC chaincode retrieveAsset: ")
+	result, err = contract.EvaluateTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Evaluate transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+```
+
+The whole text of the program would be
+```go
+package main
+
+import (
+	"os"
+
+	fpc "github.com/hyperledger-labs/fabric-private-chaincode/client_sdk/go/pkg/gateway"
+	"github.com/hyperledger-labs/fabric-private-chaincode/integration/client_sdk/go/utils"
+	"github.com/hyperledger/fabric-sdk-go/pkg/gateway"
+	"github.com/hyperledger/fabric/common/flogging"
+)
+
+var (
+	network  *gateway.Network
+	contract fpc.Contract
+	err      error
+	logger   = flogging.MustGetLogger("helloworld")
+)
+
+func main() {
+	ccID := os.Getenv("CC_ID")
+	logger.Infof("Environment variable CC_ID: %v", ccID)
+	// get network
+	network, err = utils.SetupNetwork("mychannel")
+
+	// Get FPC Contract
+	contract := fpc.GetContract(network, ccID)
+
+	// Invoke FPC Chaincode storeAsset
+	logger.Infof("--> Invoke FPC chaincode storeAsset: ")
+	result, err := contract.SubmitTransaction("storeAsset", "asset1", "100")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+	
+	// Invoke FPC Chaincode retrieveAsset
+	logger.Infof("--> Invoke FPC chaincode retrieveAsset: ")
+	result, err = contract.SubmitTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Submit transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+	
+	// Evaluate FPC Chaincode retrieveAsset
+	logger.Infof("--> Evaluate FPC chaincode retrieveAsset: ")
+	result, err = contract.EvaluateTransaction("retrieveAsset", "asset1")
+	if err != nil {
+		logger.Fatalf("Failed to Evaluate transaction: %v", err)
+	}
+	logger.Infof("--> Result: %s", string(result))
+}
+```
+* Now we have to test our code. Copy the test.sh bash script from helloworld to this directory. We are going to make some changes to this script. The first changes that need to be made are the reference to the current directory. We have to add one more level of directory `/..` to the variables `FPC_PATH` and `FABRIC_CFG_PATH`. 
+```bash
+FPC_PATH="${SCRIPTDIR}/../../.."
+FABRIC_CFG_PATH="${SCRIPTDIR}/../../../integration/config"
+```
+We also need to change the code that invoked the smart contract
```

Comment:
While following these instructions, I thought it would be easier to modify the existing `examples/helloworld/test.sh`.

Either we just add the go app in addition to the exising cli calls, or we let the reader remove the cli calls and just add the go app. But having the cli calls and go app would also not harm I believe. WDYT?

Reply:
I think for someone new to this. having both examples separate can help understand better the similarities and the differences. If we did it in the same one, they would probably have 2 bash test scripts. The unchanged one and the go one ....

Reply:
mh - let's postpone this for a moment and focus on fixing the rest of the document. Then we come back to this and get some feedback from others. Jordan, Jeb, Mic, .... and then we just do a majority voting :)